Add project structure and dockerize application

overpathz opened the pull request:
Provide structure for move convenient project lookup and contributing.
Docker application along with needed dependencies.
Add .env-sample for example of required ENV vars for OAuth2 Github

overpathz committed:
Add project structure and dockerize application

yuriytkach commented:
Super, I like it. But I have one comment: we can create docker image of our app by executing `./gradlew bootBuildImage`, which creates an optimized docker image for SpingBoot. Therefore I would not include the Dockerfile, so not to confuse that we have several way of creating an image.

overpathz committed:
Remove Dockerfile. Use prebuilt image using gradle's bootBuildImage

overpathz commented:
Addressed comment. Thx.

One thing I didn't checked -- do we can configure building specific image under a platform?
What I mean -> with Dockerfile there was no problem, but when I built image with gradle, I got this warning:

`The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested`

It's not critical for sure, but)

yuriytkach commented:
Yes, it can be configured: https://docs.spring.io/spring-boot/gradle-plugin/packaging-oci-image.html#build-image.customization

yuriytkach commented on .env-sampe:
The name probably should be `.env-sample` :)

yuriytkach reviewed (approved):
Minor comment about the filename. Otherwise, great. Thanks!

overpathz committed:
Rename file

overpathz commented:
Renamed.

Will check your live again and try to provide atomic redis change in book controller
